Hello again.
Sorry for late reply.

I tried to build it with `stack build` and got

```
[3 of 3] Compiling Main      
                             
/home/portnov/src/haskell-gettext/exe/Main.hs:13:1: error:
    Could not find module ‘Data.Text.IO.Utf8’
    Use -v to see a list of the files searched for.
   |                         
13 | import qualified Data.Text.IO.Utf8 as Utf8
```

Maybe you need to change some library version?

--- a/exe/Main.hs
+++ b/exe/Main.hs
@@ -6,24 +6,18 @@ import qualified Language.Haskell.Exts as H
 
 import Options
 
-import System.Environment
-import System.Console.GetOpt
 import Data.Time
-import System.Locale hiding (defaultTimeLocale)
 
 import Data.Generics.Uniplate.Data
 
 -- import Distribution.Simple.PreProcess.Unlit
 
 import Data.List
-import Data.Char
 import Data.Ord
 import Data.Function (on)
-import System.FilePath
 
 import Data.Version (showVersion)
-version = undefined
--- import Paths_haskell_gettext (version)
+import Paths_haskell_gettext (version)
 
 toTranslate :: [String] -> H.ParseResult (H.Module H.SrcSpanInfo) -> [(H.SrcSpanInfo, String)]
 toTranslate f (H.ParseOk z) = nub [ (loc, s) | H.App _ (H.Var _ (H.UnQual _ (H.Ident _ x))) (H.Lit _ (H.String loc s _)) <- universeBi z, x `elem` f]
@@ -32,31 +26,31 @@ toTranslate _ _ = []
 -- Create list of messages from a single file
 formatMessages :: String -> [(H.SrcSpanInfo, String)] -> String
 formatMessages path l = concat $ map potEntry $ nubBy ((==) `on` snd) $ sortBy (comparing snd) l
-    where potEntry (l, s) = unlines [
-                             "#: " ++ showSrc l,
+    where potEntry (wl, s) = unlines [
+                             "#: " ++ showSrc wl,
                              "msgid " ++ (showStringC s),
                              "msgstr \"\"",
                              ""
                             ]
-          showSrc l = path ++ ":" ++ show (H.srcSpanStartLine (H.srcInfoSpan l)) ++ ":" ++ show (H.srcSpanStartColumn (H.srcInfoSpan l))
+          showSrc wl = path ++ ":" ++ show (H.srcSpanStartLine (H.srcInfoSpan wl)) ++ ":" ++ show (H.srcSpanStartColumn (H.srcInfoSpan wl))
 
 -- Escape a string in a C-like fashion,
 -- see https://www.ibm.com/docs/en/i/7.4?topic=literals-string
 showStringC :: String -> String
-showStringC s0 = '"' : concatMap showChar s0 ++ "\""
+showStringC s0 = '"' : concatMap showCharC s0 ++ "\""
     where
-      showChar '"' = "\\\""
-      showChar '\\' = "\\\\"
-      showChar '\n' = "\\n"
-      showChar c = return c
+      showCharC '"' = "\\\""
+      showCharC '\\' = "\\\\"
+      showCharC '\n' = "\\n"
+      showCharC c = return c
 
 
 formatPotFile :: [String] -> IO String
-formatPotFile lines = do
+formatPotFile ls = do
     time <- getZonedTime
     let timeStr = formatTime defaultTimeLocale "%F %R%z" time
     let header = formatPotHeader timeStr
-    return $ concat $ header: lines
+    return $ concat $ header: ls
   where
     formatPotHeader :: String -> String
     formatPotHeader timeStr =
@@ -78,7 +72,8 @@ formatPotFile lines = do
 
 process :: Options -> IO ()
 process Options{printVersion = True} =
-        putStrLn $ "hgettext, version " ++ (showVersion version)
+        putStrLn $ "hgettext (from haskell-gettext), version " ++
+                     (showVersion version)
 process opts
     | null (inputFiles opts) = do
         putStrLn "hgettext: missing arguments"

--- a/exe/Options.hs
+++ b/exe/Options.hs
@@ -25,7 +25,7 @@ infoOpts :: ParserInfo Options
 infoOpts = info (options <**> helper)
   ( fullDesc
   <> progDesc "Extract translatable strings from Haskell source files."
-  <> header "hello - a test for optparse-applicative" )
+  <> header "hgettext (from haskell-gettext)" )
 
 options :: Parser Options
 options = Options <$> inputs <*> outfile <*> kwsDef <*> version

--- a/haskell-gettext.cabal
+++ b/haskell-gettext.cabal
@@ -62,7 +62,9 @@ library
 executable hgettext
   import:            shared
   main-is:           Main.hs
-  other-modules:     Options
+  autogen-modules:   Paths_haskell_gettext
+  other-modules:     Options,
+                     Paths_haskell_gettext
   build-depends:     filepath >= 1.4 && < 1.6,
                      haskell-src-exts >= 1.18 && < 1.24,
                      old-locale >= 1.0 && < 1.1,

--- a/test/shelltest/options/options.test
+++ b/test/shelltest/options/options.test
@@ -25,3 +25,7 @@ $ hgettext Menu.hs && cat messages.po
 # 7. Read strings (user defined keyword).
 $ hgettext --keyword __s Menu.hs && cat messages.po
 > /User defined/
+
+# 8. Show version.
+$ hgettext --version
+> /(from haskell-gettext)/

--- a/test/shelltest/translate/Menu.hs
+++ b/test/shelltest/translate/Menu.hs
@@ -1,0 +1,1 @@
+test = __s "First string."

--- a/test/shelltest/translate/generate.test
+++ b/test/shelltest/translate/generate.test
@@ -1,0 +1,3 @@
+# 1. Update a translation
+$ hgettext -k __s Menu.hs && cat messages.po
+> /First string./

